src/hip_module.cpp

```diff
@@ -1326,7 +1326,8 @@ hipError_t ihipOccupancyMaxActiveBlocksPerMultiprocessor(
     size_t numWavefronts = (blockSize + wavefrontSize - 1) / wavefrontSize;
 
     size_t availableVGPRs = (prop.regsPerBlock / wavefrontSize / simdPerCU);
-    size_t vgprs_alu_occupancy = simdPerCU * std::min(maxWavesPerSimd, availableVGPRs / usedVGPRS);
+    size_t vgprs_alu_occupancy = simdPerCU * ((usedVGPRS == 0) ? maxWavesPerSimd
```

Comment:
LGTM, otherwise excessive brackets.

Reply:
I actually used the bracket format from the SGPR check a few lines down :p
Should I change both?

Reply:
Actually, after removing I get gcc errors:

>/home/research/code/HIP/src/hip_module.cpp:1329:63: warning: operator '?:' has lower precedence than '*'; '*' will be evaluated first [-Wparentheses]
>    size_t vgprs_alu_occupancy = simdPerCU * (usedVGPRS == 0) ? maxWavesPerSimd

looks like we need to keep them

Reply:
There were other (inner) brackets in mind, actually, smth. like:
```c++
size_t vgprs_alu_occupancy = simdPerCU * (usedVGPRS ? std::min(maxWavesPerSimd, availableVGPRs / usedVGPRS) : maxWavesPerSimd);
```

Reply:
gotcha